server/src/main/java/org/opensearch/gateway/remote/RemoteClusterStateService.java

```diff
@@ -1292,6 +1347,7 @@ public ClusterState getClusterStateForManifest(
                 includeEphemeral ? manifest.getIndicesRouting() : emptyList(),
                 includeEphemeral && manifest.getHashesOfConsistentSettings() != null,
                 includeEphemeral ? manifest.getClusterStateCustomMap() : emptyMap(),
+                manifest.getDiffManifest() != null,
```

Comment:
We are supposed to send true only if includeEphemeral is `true` and routingdiff is present in manifest.
Change the condition to following
```
includeEphemeral ? manifest.getDiffManifest().getIndicesRoutingDiffPath() != null : false
```